core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPreferWritePartitioning.java

```diff
@@ -93,6 +104,53 @@ public void testDoesNotFire()
                 .doesNotFire();
     }
 
+    @Test
+    public void testThreshold()
+    {
+        PlanNodeStatsEstimate stats = PlanNodeStatsEstimate.builder()
+                .addSymbolStatistics(ImmutableMap.of(new Symbol("col_one"), new SymbolStatsEstimate(0, 0, 0, 0, 1)))
+                .build();
+
+        assertPreferredPartitioning(Optional.of(new PartitioningScheme(
+                Partitioning.create(FIXED_HASH_DISTRIBUTION, ImmutableList.of(new Symbol("col_one"))),
+                ImmutableList.of(new Symbol("col_one")))))
+                .withSession(SESSION_WITH_PREFERRED_PARTITIONING_THRESHOLD)
+                .overrideStats("mock", stats)
+                .doesNotFire();
+        // Below minimum number of partition to fire.
+    }
+
+    @Test
+    public void testThresholdWithHighNullFraction()
+    {
+        PlanNodeStatsEstimate stats = PlanNodeStatsEstimate.builder()
+                .addSymbolStatistics(ImmutableMap.of(new Symbol("col_one"), new SymbolStatsEstimate(0, 0, .5, 0, 1)))
+                .build();
+
+        assertPreferredPartitioning(Optional.of(new PartitioningScheme(
+                Partitioning.create(FIXED_HASH_DISTRIBUTION, ImmutableList.of(new Symbol("col_one"))),
+                ImmutableList.of(new Symbol("col_one")))))
+                .withSession(SESSION_WITH_PREFERRED_PARTITIONING_THRESHOLD)
+                .overrideStats("mock", stats)
+                .matches(SUCCESSFUL_MATCH);
+    }
+
+    @Test
+    public void testThresholdWithMultiplePartitions()
```

Comment:
Add a test with missing or `NaN` stats. When stats are missing, we should fallback to not use preferred partitioning. However, it should be possible to force using preferred partitioning (e.g threshold set to 0). This should be documented